src/module_2/module_2_groceries.ipynb

```diff
@@ -0,0 +1,3261 @@
+       "      <td>2020-04-30 14:32:19</td>\n",
+       "      <td>2020-04-30</td>\n",
+       "      <td>1</td>\n",
+       "      <td>33618860179588</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>2</th>\n",
+       "      <td>2204073066628</td>\n",
+       "      <td>62e271062eb827e411bd73941178d29b022f5f2de9d37f...</td>\n",
+       "      <td>2020-04-30 14:32:19</td>\n",
+       "      <td>2020-04-30</td>\n",
+       "      <td>1</td>\n",
+       "      <td>33618874040452</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>3</th>\n",
+       "      <td>2204073066628</td>\n",
+       "      <td>62e271062eb827e411bd73941178d29b022f5f2de9d37f...</td>\n",
+       "      <td>2020-04-30 14:32:19</td>\n",
+       "      <td>2020-04-30</td>\n",
+       "      <td>1</td>\n",
+       "      <td>33618907005060</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>4</th>\n",
+       "      <td>2204073066628</td>\n",
+       "      <td>62e271062eb827e411bd73941178d29b022f5f2de9d37f...</td>\n",
+       "      <td>2020-04-30 14:32:19</td>\n",
+       "      <td>2020-04-30</td>\n",
+       "      <td>1</td>\n",
+       "      <td>33618907005060</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "              id                                            user_id  \\\n",
+       "0  2204073066628  62e271062eb827e411bd73941178d29b022f5f2de9d37f...   \n",
+       "1  2204073066628  62e271062eb827e411bd73941178d29b022f5f2de9d37f...   \n",
+       "2  2204073066628  62e271062eb827e411bd73941178d29b022f5f2de9d37f...   \n",
+       "3  2204073066628  62e271062eb827e411bd73941178d29b022f5f2de9d37f...   \n",
+       "4  2204073066628  62e271062eb827e411bd73941178d29b022f5f2de9d37f...   \n",
+       "\n",
+       "           created_at order_date  user_order_seq      variant_id  \n",
+       "0 2020-04-30 14:32:19 2020-04-30               1  33618849693828  \n",
+       "1 2020-04-30 14:32:19 2020-04-30               1  33618860179588  \n",
+       "2 2020-04-30 14:32:19 2020-04-30               1  33618874040452  \n",
+       "3 2020-04-30 14:32:19 2020-04-30               1  33618907005060  \n",
+       "4 2020-04-30 14:32:19 2020-04-30               1  33618907005060  "
+      ]
+     },
+     "execution_count": 12,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "users_with_regorders = order_items[order_items.user_id.isin(regulars.user_id.unique())].reset_index(drop=True)\n",
+    "users_with_regorders.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "regulars_ordered = users_regulars[users_regulars[['user_id', 'variant_id']].apply(tuple, 1).isin(users_with_regorders[['user_id', 'variant_id']].apply(tuple, 1)) == True]\n",
```

Comment:
You could simply

    users_regulars.loc[users_regulars["variant_id"].isin(users_with_regorders["variant_id"])]

Reply:
But here you are comparing only variant_id. You don't know if that item was ordered by the user that added it to regular

Reply:
Oh my bad, true.

An alternate solution is to extract the unique pairs with `users_with_regorders[['user_id', 'variant_id']].unique()` and then do a merge of users_regulars with that with an inner join. I am wondering which would be faster 🤔